Bug fixes in Contrast Curves

markusbonse opened the pull request:
This is pull request fixes several bugs in the contrast curves modules.

The biggest change is related to this line:
https://github.com/majobodo/PynPoint/blob/ad9c415529668fdc5949f0040ef8456018a23037/pynpoint/util/limits.py#L172

the average of the noise apertures needs to be added if we want to compute the contrast curve. This is also stated in the original paper by Mawet et al 2014. In most cases this does not matter as the average is close to 0. However it can have quite some impact if we move to high PCA components.

For the calculation of the detection confidence we subtract it:
https://github.com/PynPoint/PynPoint/blob/45b3d2d9e42282d835477d8a3621833ce1769b60/pynpoint/util/analysis.py#L124

Thus, we have to add it for the contrast curves.

All tests finish without error on my local machine, but they crash on the CI.

Here are some plots of CC before and after the bug fix. 
BEFORE:
![Before_bug_fix](https://user-images.githubusercontent.com/19818846/145242888-316ed45a-4ab7-40f5-af65-ceb1e447c4e0.png)

AFTER:
![After_bug_fix](https://user-images.githubusercontent.com/19818846/145242839-bd284787-3d52-491b-92cd-7ec0842de009.png)

markusbonse committed:
Implement new function compute_aperture_flux_elements needed to fix
the bugs in the contrast curve code

markusbonse committed:
Fix bug in contrast curve code

markusbonse committed:
Small type fix based on test cases

markusbonse committed:
Update test cases to match fixed contrast curve

markusbonse committed:
Install fix for python3.8

markusbonse committed:
Install fix for python3.8 second try

markusbonse committed:
Another bug fix on computing the small sample correction

markusbonse committed:
added comment

markusbonse committed:
Fix negative throughput problem

markusbonse committed:
Fix testcases

markusbonse committed:
Fix Inf value problems

markusbonse committed:
Fix Inf value problems

Commit:
merged pynpoint master into local fork

Commit:
Update assertion values in the contrast curve

Commit:
Update error message test in pypeline to run with the pytest. All pytests run without errors

Commit:
Revert error message

majobodo commented:
It seems to run with python 3.7 but not with 3.8 and 3.9. Maybe this is due to the outdated dependencies in Pynpoint?

Commit:
Update requirements.txt and change pytests. The update of sklimage caused some changes in the values (mostly small differences). Only the function phase_cross_correlation needed a new parameter normalization=None in order to get a similar behaviour as before the update

Commit:
Updated more test cases for the contrast curves after dependencies update

Commit:
Changed several test values in the flux_position tests. In the update from scipy 1.5.3 to 1.6.x the minimization function was optimized. The results on the hessian minimization are still very similar with the following differences:
- the minimization runs for 35 not 25 iterations. We could change this by setting the maxiter to 25.
- the systematic error are much smaller

Commit:
Fix several deprecation warning related to numpy

Commit:
Fix deprecation warning related to skimage

Commit:
Update scikit-learn version

Commit:
Fix bug in core test

Commit:
Fix bug in core test

Commit:
Fix bug in core test

Commit:
add --collect-only to makefile, this prevents a Seg fault during building of the covarage

majobodo commented:
Tests and build works. The code coverage raises a Segfault for unknown reasons if not run with option  --collect-only. Maybe it no longer works
